<!DOCTYPE html>
<html lang="en">

  <!-- Head -->
  <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">    <!-- Metadata, OpenGraph and Schema.org -->
    
    <!-- Website verification -->
    <meta name="google-site-verification" content="">
<!-- Avoid warning on Google Chrome
        Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.
        see https://stackoverflow.com/a/75119417
    -->
    <meta http-equiv="Permissions-Policy" content="interest-cohort=()">

    <!-- Standard metadata -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Talks &amp; Service | Can  Firtina</title>
    <meta name="author" content="Can  Firtina">
    <meta name="description" content="List of talks, services, and collaborators.">


    <!-- Bootstrap & MDB -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous">

    <!-- Bootstrap Table -->
    <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.21.3/dist/bootstrap-table.min.css">

    <!-- Fonts & Icons -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

    <!-- Code Syntax Highlighting -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light">

    

    <!-- Styles -->
    
    <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;">
    
    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="canonical" href="https://canfirtina.com/talks/">

    <!-- Dark Mode -->
    

  </head>

  <!-- Body -->
  <body class="fixed-top-nav ">

    <!-- Header -->
    <header>

      <!-- Nav Bar -->
      <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
        <div class="container">
          <a class="navbar-brand title font-weight-lighter" href="/">Can <span class="font-weight-bold">Firtina </span></a>
          <!-- Navbar Toggle -->
          <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar top-bar"></span>
            <span class="icon-bar middle-bar"></span>
            <span class="icon-bar bottom-bar"></span>
          </button>

          <div class="collapse navbar-collapse text-right" id="navbarNav">
            <ul class="navbar-nav ml-auto flex-nowrap">

              <!-- About -->
              <li class="nav-item ">
                <a class="nav-link" href="/">About</a>
              </li>
              

              <!-- Other pages -->
              <li class="nav-item ">
                <a class="nav-link" href="/publications/">Publications</a>
              </li>
              <li class="nav-item active">
                <a class="nav-link" href="/talks/">Talks &amp; Service<span class="sr-only">(current)</span></a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/teaching/">Teaching</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/tools/">Tools</a>
              </li>
            </ul>
          </div>
        </div>
      </nav>

      <!-- Scrolling Progress Bar -->
      <progress id="progress" value="0">
        <div class="progress-container">
          <span class="progress-bar"></span>
        </div>
      </progress>
    </header>


    <!-- Content -->
    <div class="container mt-5">
      
        <!-- page.html -->
        <div class="post">

          <header class="post-header">
            <h1 class="post-title">Talks &amp; Service</h1>
            <p class="post-description">List of talks, services, and collaborators.</p>
          </header>

          <article>
            <hr>

<h2 id="invited-talks-and-lectures">Invited Talks and Lectures</h2>

<ul>
  <li>
    <p><strong>Accelerating AI-Based Genome Analysis via Algorithm-Architecture Co-Design</strong><br>
 Invited conference talk, <em><a href="https://pasc25.pasc-conference.org/" rel="external nofollow noopener" target="_blank">Platform for Advanced Scientific Computing (PASC) 2025, Challenges in Systems Design for Omics</a></em>, Switzerland. June 16 2025.</p>
  </li>
  <li>
    <p><strong>Fast and Efficient Genome Analysis via New Algorithms and Architectures</strong><br>
 Invited talk, <em><a href="https://team.inria.fr/genscale/" rel="external nofollow noopener" target="_blank">GenScale Team of Inria at the University of Rennes</a></em> (host: Pierre Peterlongo), Rennes, France. May 6 2025.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2025_05-inria-fast_efficient_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2025_05-inria-fast_efficient_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Enabling Fast, Accurate, and Efficient Real-Time Genomic Sequence Analysis via New Algorithms and Architectures</strong><br>
 Invited talk, <em><a href="https://www.amd.com/en/corporate/research/about.html" rel="external nofollow noopener" target="_blank">AMD</a></em> (virtual). September 13 2024.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_09-amd-accelerating_real-time_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_09-amd-accelerating_real-time_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Enabling Fast, Accurate, and Efficient Real-Time Genomic Sequence Analysis via New Algorithms and Architectures</strong><br>
 Invited talk, <em><a href="https://www.huawei.com/" rel="external nofollow noopener" target="_blank">Huawei Research Center Zurich</a></em>, Zurich, Switzerland. May 17 2024.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-huawei_zurich-accelerating_real-time_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-huawei_zurich-accelerating_real-time_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Real-Time Sequence Analysis from Raw Nanopore Electrical Signals with Accurate Hash-Based Search Mechanisms</strong><br>
 Invited talk, <em><a href="https://www.broadinstitute.org/bios/victoria-popic" rel="external nofollow noopener" target="_blank">The Broad Institute of MIT and Harvard</a></em> (host: Victoria Popic, Ph.D.), Cambridge, MA, USA. May 10 2024.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-broad-real-time_and_fuzzy_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-broad-real-time_and_fuzzy_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Real-Time Analysis of Genomic Sequences from Nanopore Electrical Signals by Fast and Accurate Hash-Based Search</strong><br>
 Invited talk, <em><a href="https://www.jax.org/people/joshy-george" rel="external nofollow noopener" target="_blank">The Jackson Laboratory (JAX)</a></em> (host: Joshy George, Ph.D.), Farmington, CT, USA. May 6 2024.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-jax-real-time_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-jax-real-time_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Real-Time Analysis of Genomic Sequences from Nanopore Electrical Signals by Fast and Accurate Hash-Based Search</strong><br>
 Invited talk, <em><a href="https://facultyprofiles.tufts.edu/lenore-cowen" rel="external nofollow noopener" target="_blank">Tufts University – Department of Computer Science</a></em> (host: Prof. Lenore J. Cowen), Medford, MA, USA. May 3 2024.<br>
 <a href="https://www.youtube.com/watch?v=NXuNuWDUPyY" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-tufts-real-time_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_05-tufts-real-time_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Introduction to Real-Time Raw Nanopore Signal Analysis: RawHash and RawHash2</strong><br>
 Invited lecture, <em><a href="https://people.sabanciuniv.edu/otastan/" rel="external nofollow noopener" target="_blank">BIO310 – Introduction to Bioinformatics</a></em> (Sabanci University, instructor: Oznur Tastan), Istanbul, Turkey. March 18 2024.<br>
 <a href="https://www.youtube.com/watch?v=8dmS37oXb5c" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_03-sabanci-real-time_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_03-sabanci-real-time_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>RawHash and RawHash2: Enabling Fast and Accurate Real-Time Analysis of Raw Nanopore Signals for Large Genomes using a Hash-Based Seeding Mechanism</strong><br>
 Invited talk, <em><a href="https://lit.eu/our-scientists/prof-birte-kehr/" rel="external nofollow noopener" target="_blank">Leibniz Institute for Immunotherapy (LIT)</a></em> (host: Prof. Birte Kehr), Regensburg, Germany. January 17 2024.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_01-lit-real-time_genome_analysis.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_01-lit-real-time_genome_analysis.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Enabling Fast, Accurate and Efficient Real-Time Genome Analysis via New Algorithms and Architectures</strong><br>
 Invited talk, <em><a href="https://www.ce.cit.tum.de/en/eda/persons/ulf-schlichtmann/" rel="external nofollow noopener" target="_blank">Technical University of Munich (TUM) – School of Computation, Information and Technology</a></em> (host: Prof. Ulf Schlichtmann), Munich, Germany. January 16 2024.<br>
 Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_01-tum-enabling_real-time_acceleration.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_01-tum-enabling_real-time_acceleration.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Enabling Fast, Accurate and Efficient Real-Time Genome Analysis via New Algorithms and Architectures</strong><br>
  Invited talk, <em><a href="https://www.huawei.com/" rel="external nofollow noopener" target="_blank">Huawei Munich Research Center</a></em>, Munich, Germany. January 15 2024.<br>
  Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_01-huawei-enabling_real-time_acceleration.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/firtina-2024_01-huawei-enabling_real-time_acceleration.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Enabling Accurate, Fast, and Memory-Efficient Genome Analysis via Efficient and Intelligent Algorithms</strong><br>
  Invited talk, <em><a href="https://eecs.berkeley.edu/" rel="external nofollow noopener" target="_blank">University of California, Berkeley – Department of Electrical Engineering and Computer Sciences</a></em> (host: Giulia Guidi), Berkeley, CA, USA. May 27 2022.<br>
  <a href="https://www.youtube.com/watch?v=5C3FdBXrSlg" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/enabling_fast_efficient_genome_analysis-firtina-2022_05_27-ucb.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/enabling_fast_efficient_genome_analysis-firtina-2022_05_27-ucb.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Enabling Accurate, Fast, and Memory-Efficient Genome Analysis via Efficient and Intelligent Algorithms</strong><br>
  Invited seminar talk, <em><a href="https://catalog.metu.edu.tr/course.php?prog=908&amp;course_code=9080590" rel="external nofollow noopener" target="_blank">BIN590 Graduate Seminar in Bioinformatics</a></em>, Graduate School of Informatics, Middle East Technical University (virtual). January 10 2022.<br>
  <a href="https://www.youtube.com/watch?v=Y4KMqvUYF-E" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/enabling_fast_efficient_genome_analysis-firtina-2022_01_10-metu-graduateseminar.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/enabling_fast_efficient_genome_analysis-firtina-2022_01_10-metu-graduateseminar.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
</ul>

<h2 id="conference-and-workshop-talks">Conference and Workshop Talks</h2>

<ul>
  <li>
    <p><strong>Rawsamble: Overlapping and Assembling Raw Nanopore Signals using a Hash-based Seeding Mechanism</strong><br>
 <em><a href="https://meetings.cshl.edu/abstracts.aspx?meet=DATA&amp;year=24" rel="external nofollow noopener" target="_blank">Cold Spring Harbor Laboratory (CSHL) Meeting, Biological Data Science</a></em>, NY, USA. November 2024.<br>
 <a href="https://youtu.be/D3-QytzMdMc" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawsamble_firtina_cshl-biological_data_science_2024.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawsamble_firtina_cshl-biological_data_science_2024.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Rawsamble: Overlapping and Assembling Raw Nanopore Signals using a Hash-based Seeding Mechanism</strong><br>
 <em><a href="https://www.iscb.org/ismb2024/programme-schedule/scientific-programme/hitseq" rel="external nofollow noopener" target="_blank">32nd Annual Conference on Intelligent Systems for Molecular Biology (ISMB)</a></em>, Montreal, QC, Canada. July 14 2024.<br>
 <a href="https://youtu.be/xdNN1ddp8EQ" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawsamble-firtina-2024-ismb.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawsamble-firtina-2024-ismb.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>ApHMM: Accelerating Profile Hidden Markov Models for Fast and Energy-Efficient Genome Analysis</strong><br>
 <em><a href="https://www.hipeac.net/2024/munich/" rel="external nofollow noopener" target="_blank">High Performance, Edge and Cloud Computing (HiPEAC) 2024</a></em>, Munich, Germany. January 17 2024.<br>
 <a href="https://youtu.be/a8RFca-jXPk" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/aphmm-firtina-2024-hipeac.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/aphmm-firtina-2024-hipeac.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>RawHash: Enabling Fast and Accurate Real-Time Analysis of Raw Nanopore Signals for Large Genomes</strong><br>
 <em><a href="https://www.iscb.org/ismbeccb2023-programme/tracks/hitseq" rel="external nofollow noopener" target="_blank">31st Annual ISMB / 22nd ECCB</a></em>, Lyon, France. July 25 2023.<br>
 <a href="https://youtu.be/ti0M6TvRkTs" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawhash-firtina-2023-ismbeccb.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawhash-firtina-2023-ismbeccb.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>RawHash: Enabling Fast and Accurate Real-Time Analysis of Raw Nanopore Signals for Large Genomes</strong><br>
 <em><a href="https://aacbb-workshop.github.io/" rel="external nofollow noopener" target="_blank">5th Workshop on Accelerator Architecture in Computational Biology and Bioinformatics (AACBB)</a></em>, Orlando, FL, USA. June 18 2023.<br>
 <a href="https://youtu.be/-7_pFwriK9c" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawhash-firtina-2023_06_18-aacbb.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/%7Efirtinac/pub/rawhash-firtina-2023_06_18-aacbb.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>BLEND: A Fast, Memory-Efficient and Accurate Mechanism to Find Fuzzy Seed Matches in Genome Analysis</strong><br>
 <em><a href="http://recomb2023.bilkent.edu.tr/index.html" rel="external nofollow noopener" target="_blank">27th Annual RECOMB</a></em>, Istanbul, Turkey. April 19 2023.<br>
 <a href="https://youtu.be/k9NzGwaF_mE" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/blend-firtina-2023_04_19-recomb.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/blend-firtina-2023_04_19-recomb.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>AirLift: A Fast and Comprehensive Technique for Remapping Alignments between Reference Genomes</strong><br>
 <em><a href="http://bioinformatics.csu.edu.cn/APBC2023/" rel="external nofollow noopener" target="_blank">21st Asia–Pacific Bioinformatics Conference (APBC)</a></em>, Changsha, China. April 16 2023.<br>
 <a href="https://youtu.be/nJKJK15t5YM" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/airlift-firtina-2023_04_16-apbc.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/airlift-firtina-2023_04_16-apbc.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
  <li>
    <p><strong>Apollo: A Sequencing-Technology-Independent, Scalable and Accurate Assembly Polishing Algorithm</strong><br>
 <em>ExaBiome / PASSION / SAFARI Workshop on Architectures and HPC for Genomics</em> (virtual). May 26 2021.<br>
 <a href="https://people.ee.ethz.ch/~firtinac/pub/apollo-firtina-2021_05_26-workshop.mp4" rel="external nofollow noopener" target="_blank">Video</a> · Slides (<a href="https://people.ee.ethz.ch/~firtinac/pub/apollo-firtina-2021_05_26-workshop.pdf" rel="external nofollow noopener" target="_blank">PDF</a> / <a href="https://people.ee.ethz.ch/~firtinac/pub/apollo-firtina-2021_05_26-workshop.pptx" rel="external nofollow noopener" target="_blank">PPT</a>)</p>
  </li>
</ul>

<hr>

<hr>
<h2 id="services--activities">Services &amp; Activities</h2>

<ul>
  <li>Reviewer/sub-reviewer for RECOMB, RECOMB-Seq, Bioinformatics, BMC Bioinformatics, MICRO, ISCA, HPCA, ASPLOS, PACT, IEEE Micro, FAST, DSN, IEEE Transactions on Computers, ISCAS, ISPASS, USENIX ATC.</li>
  <li>Organized <a href="https://safari.ethz.ch/recomb23-arch-workshop/" rel="external nofollow noopener" target="_blank">BIO-Arch: Workshop on Hardware Acceleration of Bioinformatics Workloads</a>, jointly held with <a href="http://recomb2023.bilkent.edu.tr/index.html" rel="external nofollow noopener" target="_blank">RECOMB 2023</a>.</li>
  <li>Organized the <a href="https://safari.ethz.ch/recomb23-arch-workshop/panel/" rel="external nofollow noopener" target="_blank">Panel on Hardware Acceleration of Bioinformatics Workloads</a> at <a href="http://recomb2023.bilkent.edu.tr/program.html" rel="external nofollow noopener" target="_blank">RECOMB 2023</a>.</li>
  <li>Organized the ExaBiome/PASSION/SAFARI Workshop on Architectures and HPC for Genomics workshop with <a href="https://giuliaguidi.github.io" rel="external nofollow noopener" target="_blank">Giulia Guidi</a>. The workshop included 10 presenters from <a href="https://sites.google.com/lbl.gov/exabiome" rel="external nofollow noopener" target="_blank">ExaBiome</a>, <a href="https://passion.lbl.gov" rel="external nofollow noopener" target="_blank">PASSION</a>, and <a href="https://safari.ethz.ch" rel="external nofollow noopener" target="_blank">SAFARI</a>. May 26 2022, Virtual.</li>
</ul>

<hr>
<h2 id="collaborators">Collaborators</h2>
<ul>
  <li>Onur Mutlu (ETH Zurich)</li>
  <li>Can Alkan (Bilkent University)</li>
  <li>Damla Senol Cali (Bionano Genomics)</li>
  <li>Jeremie S. Kim (Apple)</li>
  <li>Juan Gomez-Luna (NVIDIA)</li>
  <li>Mohammad Sadrosadati (ETH Zurich)</li>
  <li>Nika Mansouri Ghiasi (ETH Zurich)</li>
  <li>Joel Lindegger (ETH Zurich)</li>
  <li>Gagandeep Singh (AMD)</li>
  <li>Haiyu Mao (ETH Zurich)</li>
  <li>Taha Shahroodi (TU Delft)</li>
  <li>Andre Kahles (ETH Zurich)</li>
  <li>Harun Mustafa (ETH Zurich)</li>
  <li>Giulia Guidi (Cornell University)</li>
  <li>Sreenivas Subramoney (Intel Labs)</li>
  <li>Gurpreet S. Kalsi (Intel Labs)</li>
  <li>Kamlesh Pillai (Intel Labs)</li>
  <li>Mohammed Alser (ETH Zurich)</li>
  <li>Nour Almadhoun Alserr (ETH Zurich)</li>
  <li>Jisung Park (POSTECH)</li>
  <li>Konstantinos Kanellopoulos (ETH Zurich)</li>
  <li>Saugata Ghose (University of Illinois Urbana-Champaign)</li>
  <li>Serghei Mangul (USC)</li>
  <li>A. Ercument Cicek (Bilkent University)</li>
  <li>Oznur Tastan (Sabanci University)</li>
  <li>Levent Onural (Bilkent University)</li>
  <li>Ziv Bar-Joseph (Carnegie Mellon University)</li>
</ul>

          </article>

        </div>

      
    </div>

    <!-- Footer -->
    <!--    
    <footer class="fixed-bottom">
      <div class="container mt-0">
      </div>
    </footer> -->

    <!-- JavaScripts -->
    <!-- jQuery -->
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

    <!-- Bootsrap & MDB scripts -->
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script>

    <!-- Masonry & imagesLoaded -->
  <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
  <script defer src="/assets/js/masonry.js" type="text/javascript"></script>
    
  <!-- Medium Zoom JS -->
  <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script>
  <script defer src="/assets/js/zoom.js"></script>

  <!-- Bootstrap Table -->
  <script defer src="https://unpkg.com/bootstrap-table@1.21.3/dist/bootstrap-table.min.js"></script>

  <!-- Load Common JS -->
  <script src="/assets/js/no_defer.js"></script>
  <script defer src="/assets/js/common.js"></script>
  <script defer src="/assets/js/copy_code.js" type="text/javascript"></script>

    
  <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
  <script async src="https://badge.dimensions.ai/badge.js"></script>

    <!-- MathJax -->
  <script type="text/javascript">
    window.MathJax = {
      tex: {
        tags: 'ams'
      }
    };
  </script>
  <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script>
  <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>

    <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id="></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){ window.dataLayer.push(arguments); }
    gtag('js', new Date());
    gtag('config', '');
  </script>
    

<!-- Scrolling Progress Bar -->
<script type="text/javascript">
  /*
   * This JavaScript code has been adapted from the article 
   * https://css-tricks.com/reading-position-indicator/ authored by Pankaj Parashar, 
   * published on the website https://css-tricks.com on the 7th of May, 2014.
   * Couple of changes were made to the original code to make it compatible 
   * with the `al-foio` theme.
   */
  const progressBar = $("#progress");
  /*
   * We set up the bar after all elements are done loading.
   * In some cases, if the images in the page are larger than the intended
   * size they'll have on the page, they'll be resized via CSS to accomodate
   * the desired size. This mistake, however, breaks the computations as the
   * scroll size is computed as soon as the elements finish loading.
   * To account for this, a minimal delay was introduced before computing the
   * values.
   */
  window.onload = function () {
    setTimeout(progressBarSetup, 50);
  };
  /*
   * We set up the bar according to the browser.
   * If the browser supports the progress element we use that.
   * Otherwise, we resize the bar thru CSS styling
   */
  function progressBarSetup() {
    if ("max" in document.createElement("progress")) {
      initializeProgressElement();
      $(document).on("scroll", function() {
        progressBar.attr({ value: getCurrentScrollPosition() });
      });
      $(window).on("resize", initializeProgressElement);
    } else {
      resizeProgressBar();
      $(document).on("scroll", resizeProgressBar);
      $(window).on("resize", resizeProgressBar);
    }
  }
  /*
   * The vertical scroll position is the same as the number of pixels that
   * are hidden from view above the scrollable area. Thus, a value > 0 is
   * how much the user has scrolled from the top
   */
  function getCurrentScrollPosition() {
    return $(window).scrollTop();
  }

  function initializeProgressElement() {
    let navbarHeight = $("#navbar").outerHeight(true);
    $("body").css({ "padding-top": navbarHeight });
    $("progress-container").css({ "padding-top": navbarHeight });
    progressBar.css({ top: navbarHeight });
    progressBar.attr({
      max: getDistanceToScroll(),
      value: getCurrentScrollPosition(),
    });
  }
  /*
   * The offset between the html document height and the browser viewport
   * height will be greater than zero if vertical scroll is possible.
   * This is the distance the user can scroll
   */
  function getDistanceToScroll() {
    return $(document).height() - $(window).height();
  }

  function resizeProgressBar() {
    progressBar.css({ width: getWidthPercentage() + "%" });
  }
  // The scroll ratio equals the percentage to resize the bar
  function getWidthPercentage() {
    return (getCurrentScrollPosition() / getDistanceToScroll()) * 100;
  }
</script>

  </body>
</html>
